internal/exec/stages/disks/partitions.go

```diff
@@ -317,11 +320,134 @@ func (p PartitionList) Swap(i, j int) {
 	p[i], p[j] = p[j], p[i]
 }
 
+func blockDevHeld(blockDev string) (bool, error) {
+	blockDevResolved, err := filepath.EvalSymlinks(blockDev)
+	if err != nil {
+		return false, fmt.Errorf("failed to resolve %q: %v", blockDev, err)
+	}
+	_, blockDevNode := filepath.Split(blockDevResolved)
+
+	holdersDir := fmt.Sprintf("/sys/class/block/%s/holders/", blockDevNode)
+	entries, err := os.ReadDir(holdersDir)
+	if err != nil {
+		return false, fmt.Errorf("failed to retrieve holders of %q: %v", blockDev, err)
+	}
+	return len(entries) > 0, nil
+}
+
+func blockDevMounted(blockDev string) (bool, error) {
+	blockDevResolved, err := filepath.EvalSymlinks(blockDev)
+	if err != nil {
+		return false, fmt.Errorf("failed to resolve %q: %v", blockDev, err)
+	}
+
+	mounts, err := os.Open("/proc/mounts")
+	if err != nil {
+		return false, fmt.Errorf("failed to open mounts: %v", err)
+	}
+	scanner := bufio.NewScanner(mounts)
```

Comment:
Rather then a scanner can we not just use os.ReadFile and string.Contains to see whats in the file?

Reply:
We need to go through the lines to find the first entry only, and we even need to resolve symlinks for normalization because `/proc/mounts` can have something like `/dev/mapper/NAME` which is a symlink to `/dev/dm-X`.